diff --git a/dist/server/authenticateRequest.js b/dist/server/authenticateRequest.js
index d9f71ad52955200e4e552c8fc2c0ed890784603a..919440ecaf75921aadddf16fdd5806190cc05813 100644
--- a/dist/server/authenticateRequest.js
+++ b/dist/server/authenticateRequest.js
@@ -3,6 +3,7 @@ import { constants, AuthStatus } from '@clerk/backend/internal';
 import { handleNetlifyCacheInDevInstance } from '@clerk/shared/netlifyCacheHandler';
 import { errorThrower } from '../utils/index.js';
 import { patchRequest } from './utils/index.js';
+import { getEvent } from '@tanstack/react-start/server';
 
 // src/server/authenticateRequest.ts
 async function authenticateRequest(request, opts) {
@@ -26,16 +27,18 @@ async function authenticateRequest(request, opts) {
     afterSignInUrl,
     afterSignUpUrl
   });
+  const event = getEvent();
   const locationHeader = requestState.headers.get(constants.Headers.Location);
   if (locationHeader) {
+    console.log('process.env', process.env)
+    console.log('event.context', event.context)
     console.log("is on netlify process.env.NETLIFY", process.env.NETLIFY);
-    console.log("is on netlify import.meta.env.NETLIFY", import.meta.env.NETLIFY);
     handleNetlifyCacheInDevInstance({
       locationHeader,
       requestStateHeaders: requestState.headers,
       publishableKey: requestState.publishableKey,
       // eslint-disable-next-line turbo/no-undeclared-env-vars
-      netlifyEnv: import.meta.env.NETLIFY
+      netlifyEnv: true
     });
     throw new Response(null, { status: 307, headers: requestState.headers });
   }
