diff --git a/dist/server/authenticateRequest.js b/dist/server/authenticateRequest.js
index b3662cd26117ee55a8e9ddacf1810aa55e5c1b15..4b3e61cb6421f5817c4b99ae2305d9453e74d864 100644
--- a/dist/server/authenticateRequest.js
+++ b/dist/server/authenticateRequest.js
@@ -27,11 +27,15 @@ async function authenticateRequest(request, opts) {
   });
   const hasLocationHeader = requestState.headers.get("location");
   if (hasLocationHeader) {
+    requestState.headers.set('Netlify-Vary', 'cookie=__client|__session');
+    requestState.headers.set('X-Debug-Request', 'true');
+    requestState.headers.set('Cache-Control', 'no-store');
     throw new Response(null, { status: 307, headers: requestState.headers });
   }
   if (requestState.status === AuthStatus.Handshake) {
     throw errorThrower.throw("Clerk: unexpected handshake without redirect");
   }
+  requestState.headers.set('Netlify-Vary', 'cookie=__client|__session');
   return requestState;
 }
 
