diff --git a/dist/server/authenticateRequest.js b/dist/server/authenticateRequest.js
index 0d4709a65c032bc5dfc3595fe9aeeb94aae28ea6..d61414f4ed67eac33a4d09683313c3fa9f481796 100644
--- a/dist/server/authenticateRequest.js
+++ b/dist/server/authenticateRequest.js
@@ -27,6 +27,13 @@ async function authenticateRequest(request, opts) {
   });
   const hasLocationHeader = requestState.headers.get("location");
   if (hasLocationHeader) {
+    // console.log('hasLocationHeader', hasLocationHeader)
+    // const hasHandshakeQueryParam = hasLocationHeader.includes('__clerk_handshake');
+    // if (!hasHandshakeQueryParam) {
+    //   const url = new URL(hasLocationHeader);
+    //   url.searchParams.append('NETLIFY_CACHE_BUST_PARAM', Date.now().toString());
+    //   requestState.headers.set('Location', url.toString());
+    // }
     throw new Response(null, { status: 307, headers: requestState.headers });
   }
   if (requestState.status === AuthStatus.Handshake) {
